File: VB/MyDialogViewModel.vb
Imports DevExpress.Mvvm
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace DialogTest
    Public Interface IDialogViewModel
        Property DialogWidth() As Integer
        Property DialogHeight() As Integer
        Property DialogTitle() As String
        Property CommandsSource() As IEnumerable(Of UICommand)
    End Interface

    Public Class MyDialogViewModel
        Implements IDialogViewModel

        Public Sub New()
            DialogTitle = "Print dialog"
            DialogWidth = 640
            DialogHeight = 480
            CommandsSource = New List(Of UICommand) From { _
                New UICommand With { _
                    .Id = System.Windows.MessageBoxResult.Cancel, _
                    .Caption = "Cancel", _
                    .IsCancel = True, _
                    .Tag = System.Windows.MessageBoxResult.Cancel _
                }, _
                New UICommand With { _
                    .Caption = "Print", _
                    .IsDefault = True, _
                    .Command = New DelegateCommand(AddressOf Print), _
                    .Tag = System.Windows.MessageBoxResult.OK _
                } _
            }
        End Sub

        #Region "IDialogViewModel"
        Public Property DialogWidth() As Integer Implements IDialogViewModel.DialogWidth
        Public Property DialogHeight() As Integer Implements IDialogViewModel.DialogHeight
        Public Property DialogTitle() As String Implements IDialogViewModel.DialogTitle
        Public Property CommandsSource() As IEnumerable(Of UICommand) Implements IDialogViewModel.CommandsSource
        #End Region ' IDialogViewModel

        Private Sub Print()

        End Sub

    End Class
End Namespace

File: CS/MyDialogViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DialogTest
{
    public interface IDialogViewModel
    {
        int DialogWidth { get; set; }
        int DialogHeight { get; set; }
        string DialogTitle { get; set; }
        IEnumerable<UICommand> CommandsSource { get; set; }
    }

    public class MyDialogViewModel : IDialogViewModel
    {
        public MyDialogViewModel()
        {
            DialogTitle = "Print dialog";
            DialogWidth = 640;
            DialogHeight = 480;
            CommandsSource = new List<UICommand>
            {
                new UICommand
                {
                    Id = System.Windows.MessageBoxResult.Cancel,
                    Caption = "Cancel",
                    IsCancel = true,
                    Tag =  System.Windows.MessageBoxResult.Cancel

                },
                new UICommand
                {
                    Caption = "Print",
                    IsDefault = true,
                    Command = new DelegateCommand(Print),
                    Tag = System.Windows.MessageBoxResult.OK
                }
            };
        }

        #region IDialogViewModel
        public int DialogWidth { get; set; }
        public int DialogHeight { get; set; }
        public string DialogTitle { get; set; }
        public IEnumerable<UICommand> CommandsSource { get; set; }
        #endregion IDialogViewModel

        private void Print()
        {

        }

    }
}
